code/datums/records.dm

```diff
@@ -0,0 +1,194 @@
+// Generic data stored in record
+/datum/record
+	var/id
+	var/notes = "No notes found."
+
+	var/cmp_field = "id"
+	var/list/excluded_fields
+
+/datum/record/New()
+	..()
+	var/tmp_ex = excluded_fields
+	excluded_fields = list()
+	for(var/f in tmp_ex)
+		excluded_fields[f] = f
+
+/datum/record/proc/Copy(var/datum/copied)
+	if(!copied)
+		copied = new src.type()
+	var/exclusions = (SSrecords.excluded_fields | src.excluded_fields)
+	for(var/variable in src.vars)
+		if(exclusions[variable]) continue
+		if(istype(src.vars[variable], /datum/record) || istype(src.vars[variable], /list))
+			copied.vars[variable] = src.vars[variable].Copy()
+		else
+			copied.vars[variable] = src.vars[variable]
+	return copied
+
+/datum/record/proc/Listify(var/deep = 1, var/list/excluded = list(), var/list/to_update) // Mostyl to support old things or to use with serialization
+	var/list/record
+	if(!to_update)
+		. = record = list()
```

Comment:
Personally I nerve seen this syntax before. Confusing. I know what it does now

Reply:
This is same as:
```
record = list()
. = record
```